tests/rpc/rpc_01-system_test.go

```diff
@@ -0,0 +1,181 @@
+			method:      "system_version",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_chain",
+			method:      "system_chain",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_properties",
+			method:      "system_properties",
+			skip:        true,
+		},
+		{
+			description: "test system_health",
+			method:      "system_health",
+			expected: modules.SystemHealthResponse{
+				Health: common.Health{
+					Peers:           2,
+					IsSyncing:       false,
+					ShouldHavePeers: true,
+				},
+			},
+			skip: false,
+		},
+		{
+			description: "test system_peers",
+			method:      "system_peers",
+			expected: modules.SystemPeersResponse{
+				Peers: []common.PeerInfo{},
+			},
+			skip: false,
+		},
+		{
+			description: "test system_network_state",
+			method:      "system_networkState",
+			expected: modules.SystemNetworkStateResponse{
+				NetworkState: common.NetworkState{
+					PeerID: "",
+				},
+			},
+			skip: false,
+		},
+		{ //TODO
+			description: "test system_addReservedPeer",
+			method:      "system_addReservedPeer",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_removeReservedPeer",
+			method:      "system_removeReservedPeer",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_nodeRoles",
+			method:      "system_nodeRoles",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_accountNextIndex",
+			method:      "system_accountNextIndex",
+			skip:        true,
+		},
+	}
+
+	t.Log("going to start gossamer")
+
+	localPidList, err := StartNodes(t, make([]*exec.Cmd, 1))
+
+	//use only first server for tests
+	require.Nil(t, err)
+
+	time.Sleep(time.Second) // give server a second to start
+
+	for _, test := range testsCases {
+		t.Run(test.description, func(t *testing.T) {
+			if test.skip {
+				t.Skip("RPC endpoint not yet implemented")
+				return
+			}
+
+			respBody, err := PostRPC(t, test.method, "http://"+GOSSAMER_NODE_HOST+":"+currentPort, "{}")
+			require.Nil(t, err)
+
+			target := DecodeRPC(t, respBody, test.method)
+
+			require.NotNil(t, target)
+
+			t.Log("Will start assertion for ", "target", target, "type", reflect.TypeOf(target))
+
+			switch v := target.(type) {
+
+			case *modules.SystemHealthResponse:
+				t.Log("Will assert SystemHealthResponse", "target", target)
+
+				require.Equal(t, test.expected.(modules.SystemHealthResponse).Health.IsSyncing, v.Health.IsSyncing)
+				require.Equal(t, test.expected.(modules.SystemHealthResponse).Health.ShouldHavePeers, v.Health.ShouldHavePeers)
+				require.GreaterOrEqual(t, test.expected.(modules.SystemHealthResponse).Health.Peers, v.Health.Peers)
+
+			case *modules.SystemNetworkStateResponse:
+				t.Log("Will assert SystemNetworkStateResponse", "target", target)
+
+				require.NotNil(t, v.NetworkState)
+				require.NotNil(t, v.NetworkState.PeerID)
+
+			case *modules.SystemPeersResponse:
+				t.Log("Will assert SystemPeersResponse", "target", target)
```

Comment:
do we need logs here for everything? seems a bit excessive

Reply:
Well, this logs help me to make sure that we are actually getting into the case statement. I am going to remove this as part of the next iteration when the tests will be more robust and I am more confident on its behavior. Right now the logs are important since we are building up.

Reply:
raised  #807 to cover it